addon/services/m3-schema-manager.js

```diff
@@ -35,7 +46,30 @@ export default class SchemaManager extends Service {
    * @returns {Object}
    */
   computeNestedModel(key, value, modelName, schemaInterface) {
-    return this.get('schema').computeNestedModel(key, value, modelName, schemaInterface);
+    let schema = this.get('schema');
+    if (this.useComputeAttribute()) {
+      let computedValue = schema.computeAttribute(key, value, modelName, schemaInterface);
+      let valueType = schemaTypesInfo.get(computedValue);
+      if (valueType === NESTED) {
+        return computedValue;
+      }
+    } else {
+      return schema.computeNestedModel(key, value, modelName, schemaInterface);
+    }
+  }
+
+  computeAttribute(key, value, modelName, schemaInterface) {
+    return this.get('schema').computeAttribute(key, value, modelName, schemaInterface);
+  }
+
+  useComputeAttribute() {
+    let schema = this.get('schema');
+    let isComputeAttributeReferenceMissing = !schema.computeAttributeReference;
+    let defaultPrototype = DefaultSchema.prototype;
+    let usingDefaultSchemaMethods =
+      defaultPrototype.computeAttributeReference === schema.computeAttributeReference &&
+      defaultPrototype.computeNestedModel === schema.computeNestedModel;
+    return isComputeAttributeReferenceMissing || usingDefaultSchemaMethods;
```

Comment:
It makes more sense to me to check against `computeAttribute` being implemented than the inverse

Reply:
Fixed